### ISSUE TITLE
Tag a version to allow releasing on Homebrew

### ISSUE THREAD

**Issue by modocache** [facebook/xcbuild#66]
It'd be awesome if I could `brew install xcbuild`. I'd be happy to submit a Homebrew formula pull request for xcbuild, but it'll need a tagged version in order for me to do so. How about tagging the current `HEAD` of `master` as `0.1.0`?

**Comment by modocache** [comment]
I'm no Homebrew expert, but the following formula works for me:

```rb
# Formula/xcbuild.rb

class Xcbuild < Formula
  desc "Xcode-compatible build tool."
  homepage "https://github.com/modocache/xcbuild/"
  url "https://github.com/modocache/xcbuild.git", :tag => "0.1.0",
                                                  :shallow => false

  depends_on "cmake" => :build
  depends_on "ninja" => :build

  def install
    system "make"
    system "make", "install", "PREFIX=#{prefix}"
    bin.install_symlink "#{prefix}/usr/bin/xcbuild"
  end

  test do
    system "#{bin}/xcbuild", "-showsdks"
  end
end
```

Thoughts, @facebook/team-xcbuild?

**Comment by grp** [comment]
The issue I was expecting with this is that the tools in xcbuild conflict with the system tools. For example, xcbuild has an implementation of `plutil` that's generally compatible with the system one, but not one that I'd suggest putting first in your `PATH`. If I understand your formula right, though, it's installing just the name `xcbuild` into the `PATH` and leaving the rest in a Homebrew-managed prefix. That looks like a good solution to avoid conflicts.

I'm not sure about tagging versions yet, but maybe it's time! I'm not sure, but there might be a way to avoid that by setting it up as a private or testing package at least initially.

You'll also want to follow the discussion on c25336b50b7735c50a692f2453b6dded435ceb14 — since your draft formula, xcbuild now installs into a developer root instead of a Unix root. @strager's new approach discussed on that will probably be helpful here.

**Comment by modocache** [comment]
> I'm not sure about tagging versions yet, but maybe it's time!

For a version like v0.1.0, I figure why not? You could even tag v0.0.1 if you'd like to make extra clear that you're not worrying about semver yet (or potentially ever).

> I'm not sure, but there might be a way to avoid that by setting it up as a private or testing package at least initially.

Sure, I'll look into this after updating my formula for the new approach (once it's ready). Maybe we can create a private tap or something.

**Comment by grp** [comment]
Sounds good to me!

**Comment by modocache** [comment]
Awesome, thanks @grp! Do you mind if I tag current `master` 08575f65f1c907e280ed4663b5461b7cfaf5cfaa as v0.1.0?

**Comment by grp** [comment]
I don't know any issues with master right now, should be good.

**Comment by modocache** [comment]
Cool, thanks! https://github.com/facebook/xcbuild/tree/0.1.0

**Comment by modocache** [comment]
I looked over the discussion on https://github.com/facebook/xcbuild/commit/c25336b50b7735c50a692f2453b6dded435ceb14, and over the related pull requests https://github.com/facebook/xcbuild/pull/73 and https://github.com/facebook/xcbuild/pull/81, but it looks like none of those changes made it into `0.1.0`. As a result, I think the following Homebrew formula suits our needs: https://github.com/modocache/homebrew-core/commit/018f97657738a26b3a571db3736c8c6974532186

It's nearly identical to the one I posted above, but with some slight fixes for `brew audit --strict --online xcbuild`. If you don't have any objections, I'll submit a pull request to Homebrew.

**Comment by grp** [comment]
Looks ok to me. My only concern is that I'm not sure how stable the `make` wrapper will be — it might be better to use `cmake` directly. But since the formula is tied to a tag, it should be fine for now and can be changed for later updates if needed.

**Comment by modocache** [comment]
Yeah, the fact that the formula is tied to a version is a big help here. Here's the formula pull request: https://github.com/Homebrew/homebrew-core/pull/3262

**Comment by grp** [comment]
Looks like it failed on the Homebrew test machines since xcbuild needs libpng. #87 will remove that requirement for OS X, but it isn't merged yet so should be easy as adding libpng to the dependencies of the formula.

**Comment by modocache** [comment]
Any thoughts on the Homebrew CI failures on 10.10 Mavericks and 10.9 Yosemite? I don't really care about supporting those, but then I'm also unsure how to add a Homebrew formula dependency on a specific OS version.

**Comment by grp** [comment]
The Mavericks failure seems to be due to an older C++ standard library. Should be easy enough to fix that with a machine to try building on; the specific error that it's failing on could probably be fixed just by not using the uniform initialization syntax (`{}`) but there's probably more after that. If you do fix this, we might be able to add a Travis check for that here too.

The Yosemite failure is strange to me — it's failing because of `libcompression.dylib`, but xcbuild only links that if it can find it. Maybe it's building against a newer SDK then trying to *run* on Yosemite, which wouldn't work since it found the library at compile time? Either way, long-term xcbuild should move onto the [open source](https://github.com/lzfse/lzfse) implementation of this rather than the OS X library.

**Comment by Palleas** [comment]
I've tried to pick up @modocache's work but I'm getting a little overwhelmed by the questions about cmake, would somebody be available to join in Homebrew/homebrew-core#5986 ? ❤️
